DASH-IF-IOP.bs.md

```diff
@@ -10,9 +10,1070 @@ Live discussion in #document-authoring on Slack.
+
+To perform the conversion, execute the following steps:
+
+<div class="algorithm">
+
+1. Calculate the number of [=media segments=] in the representation as `SegmentCount = Ceil(AsSeconds(Period@duration) / ( SegmentTemplate@duration / SegmentTemplate@timescale))`.
+1. Update the MPD.
+	1. Add a single `SegmentTemplate/SegmentTimeline` element.
+	1. Add a single `SegmentTimeline/S` element.
+	1. Set `S@t` to equal `SegmentTemplate@presentationTimeOffset`.
+	1. Set `S@d` to equal `SegmentTemplate@duration`.
+	1. Remove `SegmentTemplate@duration`.
+	1. Set `S@r` to `SegmentCount - 1`.
+
+</div>
+
+<div class="example">
+Below is an example of a [=simple addressing=] [=representation=] before conversion.
+
+<xmp highlight="xml">
+<MPD xmlns="urn:mpeg:dash:schema:mpd:2011">
+	<Period duration="PT900S">
+		<AdaptationSet>
+			<Representation>
+				<SegmentTemplate timescale="1000" presentationTimeOffset="900"
+						media="video/$Number$.m4s" initialization="video/init.mp4"
+						duration="4001" startNumber="800" />
+			</Representation>
+		</AdaptationSet>
+	</Period>
+</MPD>
+</xmp>
+
+As part of the conversion, we calculate `SegmentCount = Ceil(900 / (4001 / 1000)) = 225`.
+
+After conversion, we arrive at the following result.
+
+<xmp highlight="xml">
+<MPD xmlns="urn:mpeg:dash:schema:mpd:2011">
+	<Period duration="PT900S">
+		<AdaptationSet>
+			<Representation>
+				<SegmentTemplate timescale="1000" presentationTimeOffset="900"
+						media="video/$Number$.m4s" initialization="video/init.mp4"
+						startNumber="800">
+					<SegmentTimeline>
+						<S t="900" d="4001" r="224" />
+					</SegmentTimeline>
+				</SegmentTemplate>
+			</Representation>
+		</AdaptationSet>
+	</Period>
+</MPD>
+</xmp>
+
+Parts of the MPD structure that are not relevant for this chapter have been omitted - the above are not fully functional MPD files.
+</div>
+
+### Expanding URL template variables ### {#timing-template-variables}
+
+Expansion rules for URL template variables such as `$Time$` and `$Number` are defined by [[!MPEGDASH]].
+
+The set of string formatting suffixes used SHALL be restricted to `%0[width]d`.
+
+Note: The string format suffixes are not intended for general-purpose string formatting. Restricting it to only this single suffix enables the functionality to be implemented without a string formatting library.
+
+## Segment alignment ## {#timing-segmentalignment}
+
+[=Media segments=] are said to be aligned if the start/end points of all [=media segments=] on the [=sample timeline=] are equal in all [=representations=] that belong to the same adaptation set.
```

Comment:
> start/end points of all media segments on the sample timeline are equal

From @haudiobe: Please check MPEG-DASH – very clearly defined in clause 4. The above is not correct as it would not allow different frame rates in one adaptation set.

Reply:
If different frame rates would lead to non-aligned segments then YES they are not allowed in one adaptation set.

However, it does not necessarily follow that different frame rates would lead to non-aligned segments. You can still do it with alignment, you just cannot do it arbitrarily but have to maintain the alignment.